ee/query-service/dao/sqlite/domain.go

```diff
@@ -28,29 +28,58 @@ type StoredDomain struct {
 
 // GetDomainFromSsoResponse uses relay state received from IdP to fetch
 // user domain. The domain is further used to process validity of the response.
-// when sending login request to IdP we send relay state as URL (site url) 
-// with domainId as query parameter.   
+// when sending login request to IdP we send relay state as URL (site url)
+// with domainId or domainName as query parameter.
 func (m *modelDao) GetDomainFromSsoResponse(ctx context.Context, relayState *url.URL) (*model.OrgDomain, error) {
 	// derive domain id from relay state now
-	var domainIdStr string 
+	var domainIdStr string
+	var domainNameStr string
+	var domain *model.OrgDomain
+
 	for k, v := range relayState.Query() {
 		if k == "domainId" && len(v) > 0 {
 			domainIdStr = strings.Replace(v[0], ":", "-", -1)
 		}
+		if k == "domainName" && len(v) > 0 {
+			domainNameStr = v[0]
+		}
 	}
 
-	domainId, err := uuid.Parse(domainIdStr)
-	if err != nil {
-		zap.S().Errorf("failed to parse domain id from relay state", err)
-		return nil, fmt.Errorf("failed to parse response from IdP response")
+	if domainIdStr != "" {
+		domainId, _ := uuid.Parse(domainIdStr)
+		domain, _ = m.GetDomain(ctx, domainId)
+
 	}
 
-	domain, err := m.GetDomain(ctx, domainId)
-	if (err != nil) || domain == nil {
-		zap.S().Errorf("failed to find domain received in IdP response", err.Error())
-		return nil, fmt.Errorf("invalid credentials")
+	if domainNameStr != "" {
+
+		domain, _ = m.GetDomainByName(ctx, domainNameStr)
+
+	}
+	if domain != nil {
+		return domain, nil
 	}
 
+	return nil, fmt.Errorf("failed to find domain received in IdP response")
+}
+
+// GetDomain returns org domain for a given domain id
+func (m *modelDao) GetDomainByName(ctx context.Context, name string) (*model.OrgDomain, basemodel.BaseApiError) {
+
+	stored := StoredDomain{}
+	err := m.DB().Get(&stored, `SELECT * FROM org_domains WHERE name=$1 LIMIT 1`, name)
+
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, model.BadRequest(fmt.Errorf("invalid domain name"))
+		}
+		return nil, model.InternalError(err)
+	}
+
+	domain := &model.OrgDomain{Id: stored.Id, Name: stored.Name, OrgId: stored.OrgId}
+	if err := domain.LoadConfig(stored.Data); err != nil {
+		return domain, model.InternalError(err)
```

Comment:
Not sure what the `Data` part contains, does it make sense to return both domain and error at the same time? And the error is ignored by the caller `domain, _ = m.GetDomainByName(ctx, domainNameStr)`.

Reply:
I got this from `GetDomainByEmail` function already present. I checked `stored.Data`, it contains the SAML config stored i.e., ACS Url, EntityId and X.509 certificate. I think, it makes sense to send back the domain for further checks if there is error in loading the SAML config.

Reply:
The `SamlConfig` entries/fieds are used to prepare the request https://github.com/SigNoz/signoz/blob/ee6b290a0ca2300412427824f44487462794d1d8/ee/query-service/model/domain.go#L114-L136

Since the config is the main part of request, I think we shouldn't have the both domain and error. Or if it is fine to send them both we need to have the caller to act based on both the returned values. In this change the caller ignores the error because it checks if domain != nil. This may not be an issue in the happy flow when the jsondata is correct but we should avoid preparing invalid request.

Reply:
Sure

Reply:
I am not sure what you are referring in the above context. What I checked is that if the caller function immediately returns when an error is observed without making any use of `domain` then it's useless to return. I checked and none of the caller functions are using `domain` returned but just return when an error is observed.

Reply:
We can discuss more on https://github.com/SigNoz/signoz/pull/3421

Reply:
> I checked and none of the caller functions are using domain returned but just return when an error is observed.


This was not the case when PR was initially opened. This discussion doesn't make sense after this commit https://github.com/SigNoz/signoz/pull/3396/commits/502540921cfff9bcbbb34d42d53157fc4ed1db2d.